### PR TITLE
 Fix #4648 

### LINKED ISSUE #4648: Remove warning on about-to-be-deleted corrupt resgen output
context: https://github.com/microsoft/msbuild/pull/4644#discussion_r316363861

Currently, there are some circumstances that will make the `GenerateResource` task run `RemoveCorruptedFile`

https://github.com/microsoft/msbuild/blob/3885a205f40944174a3e14396547f13db1196b82/src/Tasks/GenerateResource.cs#L2896-L2912

That method unconditionally emits a warning that the file may be corrupted. But it then immediately deletes it (or emits another error that it couldn't delete it).

The warning seems unnecessary. We should bump it down to info and make it more descriptive, maybe something like

> Attempting to remove possibly-corrupt output file {}.

### PR DESCRIPTION
**Pull request by treymerkley**

### CODE REVIEW

File: src/Tasks/GenerateResource.cs [hunk 1]
```diff
@@ -2895,14 +2895,14 @@ private string EnsurePathIsShortEnough(string currentOutputFile, string currentO
         /// <param name="filename">Full path to file to delete</param>
         private void RemoveCorruptedFile(string filename)
         {
-            _logger.LogWarningWithCodeFromResources("GenerateResource.CorruptOutput", FileUtilities.GetFullPathNoThrow(filename));
+            _logger.LogMessageFromResources("GenerateResource.CorruptOutput. Attempting to remove possibly-corrupt output file: ", FileUtilities.GetFullPathNoThrow(filename));
```
Comment (rainersigwald) [inline src/Tasks/GenerateResource.cs]: This is changing the name of the resource, and you shouldn't need to do that.
```suggestion
            _logger.LogMessageFromResources("GenerateResource.CorruptOutput", FileUtilities.GetFullPathNoThrow(filename));
```

File: src/Tasks/GenerateResource.cs [hunk 2]
```diff
@@ -2895,14 +2895,14 @@ private string EnsurePathIsShortEnough(string currentOutputFile, string currentO
...
             try
             {
                 File.Delete(filename);
             }
             catch (Exception deleteException)
             {
-                _logger.LogWarningWithCodeFromResources("GenerateResource.DeleteCorruptOutputFailed", FileUtilities.GetFullPathNoThrow(filename), deleteException.Message);
+                _logger.LogInfoWithCodeFromResources("GenerateResource.DeleteCorruptOutputFailed. Failed to remove possibly-corrupt output file: ", FileUtilities.GetFullPathNoThrow(filename), deleteException.Message);
```
Comment (rainersigwald) [inline src/Tasks/GenerateResource.cs]: I think you can revert this entirely; if we _fail_ to delete the file we should still log a warning (that should be rare).

File: src/Tasks/Resources/Strings.resx [hunk 3]
```diff
@@ -911,7 +911,6 @@
   </data>
   <data name="GenerateResource.CorruptOutput">
     <value>MSB3555: Output file "{0}" is possibly corrupt.</value>
```
Comment (rainersigwald) [inline src/Tasks/Resources/Strings.resx]:
```suggestion
     <value>Output file "{0}" is possibly corrupt and will be deleted and recreated.</value>
```

File: src/Tasks/Resources/Strings.resx [hunk 4]
```diff
@@ -931,7 +930,6 @@
   </data>
   <data name="GenerateResource.DeleteCorruptOutputFailed">
     <value>MSB3560: Could not delete the possibly corrupt output file "{0}". {1}</value>
-    <comment>{StrBegin="MSB3560: "}</comment>
```
Comment (rainersigwald) [inline src/Tasks/Resources/Strings.resx]: If we leave this as a warning we can revert this change.

File: src/Tasks/Resources/Strings.resx (hunk 3 above)
Comment (rainersigwald) [reply src/Tasks/Resources/Strings.resx]: Making this change will then require running a build to propagate it to the localized `.xlf`s.

File: src/Tasks/GenerateResource.cs [hunk 5]
```diff
@@ -56,7 +56,7 @@ public sealed partial class GenerateResource : TaskExtension
 
 
 #region Fields
-        
+
```
Comment (rainersigwald) [inline src/Tasks/GenerateResource.cs]: Would you mind reverting the whitespace changes in this file? I think your editor (correctly) respected our `.editorconfig` but we haven't actually fixed that across the tree yet . . .

File: src/Tasks/GenerateResource.cs (hunk 1 above)
Comment (treymerkley) [reply src/Tasks/GenerateResource.cs]: I added that in because of this:

> The warning seems unnecessary. We should bump it down to info and make it more descriptive, maybe something like
>
>   ` Attempting to remove possibly-corrupt output file {}.`

How should I work that in then?

Comment (rainersigwald) [reply src/Tasks/GenerateResource.cs]: Change the contents of the string (in the `.resx` file and localizations in the various `.xlf`), not its name.

As an analogy, you're doing this:

```diff
- var generateResource_CorruptOutput = "MSB3555: Output file "{0}" is possibly corrupt.";
+ var generateResource_CorruptOutput_attempting_to_remove = "MSB3555: Output file "{0}" is possibly corrupt.";
```

instead of

```diff
- var generateResource_CorruptOutput = "MSB3555: Output file "{0}" is possibly corrupt.";
+ var generateResource_CorruptOutput = "Output file "{0}" is possibly corrupt.";
```

File: src/Tasks/GenerateResource.cs (hunk 5 above)
Comment (treymerkley) [reply src/Tasks/GenerateResource.cs]: Oh sorry, I have my text editor set to always delete trailing whitespace. Is that something that I shouldn't have?

Comment (treymerkley) [reply src/Tasks/GenerateResource.cs]: I just fixed it, I'll push it in just a second.

Comment (rainersigwald) [reply src/Tasks/GenerateResource.cs]: I have no problem with fixing trailing whitespace, but I'd like to see it separated from the contents of this commit so it's easier to understand changes in the future.

Comment (treymerkley) [reply src/Tasks/GenerateResource.cs]: wait nope, that is the issue, I think. I'll fix it after everything else in something like notepad or something.

Comment (treymerkley) [reply src/Tasks/GenerateResource.cs]: sure, I'll revert it and then push it as a new commit.

File: src/Tasks/GenerateResource.cs (hunk 2 above)
Comment (treymerkley) [reply src/Tasks/GenerateResource.cs]: okay, *should* be fixed

File: src/Tasks/GenerateResource.cs (hunk 5 above)
Comment (treymerkley) [reply src/Tasks/GenerateResource.cs]: should be fixed.

File: src/Tasks/Resources/Strings.resx (hunk 4 above)
Comment (treymerkley) [reply src/Tasks/Resources/Strings.resx]: okay, so just add back in the line `<comment>{StrBegin="MSB3560: "}</comment>` below 932?

Comment (rainersigwald) [reply src/Tasks/Resources/Strings.resx]: Yes.

Comment (treymerkley) [reply src/Tasks/Resources/Strings.resx]: *should* be fixed

File: src/Tasks/Resources/Strings.resx [hunk 6]
```diff
@@ -910,8 +910,7 @@
     <comment>{StrBegin="MSB3554: "}</comment>
   </data>
   <data name="GenerateResource.CorruptOutput">
-    <value>MSB3555: Output file "{0}" is possibly corrupt.</value>
-    <comment>{StrBegin="MSB3555: "}</comment>
+     <value>Output file "{0}" is possibly corrupt and will be deleted and recreated.</value>
```
Comment (rainersigwald) [inline src/Tasks/Resources/Strings.resx]:
```suggestion
    <value>Output file "{0}" is possibly corrupt and will be deleted and recreated.</value>
```
